### PR TITLE
PMM-6153 Add local rss. Fallback on local rss if cannot connect to pe…

### PR DESCRIPTION
**Pull request by askomorokhov**
…rcona blog.

https://jira.percona.com/browse/PMM-6153

FB: https://github.com/Percona-Lab/pmm-submodules/pull/906
related PR: https://github.com/percona/pmm-server-packaging/pull/140

### CODE REVIEW

File: nginx.conf [hunk 1]
```diff
@@ -172,11 +172,16 @@
       proxy_set_header Connection "";
     }
 
-    # proxy requests to the Percona news rss feed
+    # proxy requests to the Percona's blog feed
+    # fallback to local rss if pmm-server is isolated from internet.
+    # https://jira.percona.com/browse/PMM-6153
+    root /usr/share/pmm-server;
```
Comment (atymchuk) [inline nginx.conf:178]: Shouldn't we put this `root` inside `location`? Or, if this is global, let's move it to the top of the config.

Comment (askomorokhov) [reply nginx.conf:178]: I’d like to put it into location as well. Check different solution, but it works only in such way.

Comment (AlekSi) [reply nginx.conf]: We definitely should not use `root`. I think `alias` inside `location` should work. Please try it.

File: nginx.conf [hunk 2]
```diff
@@ -172,11 +172,16 @@
...
     location = /percona-blog/feed {
       auth_request off;
 
-      proxy_pass https://www.percona.com/blog/feed/;
+      set $feed https://www.percona.com/blog/feed/;
+      proxy_pass $feed;
       proxy_set_header User-Agent "$http_user_agent pmm-server/2.x";
-    }     
+      error_page 500 502 503 504 /local-rss.xml;
```
Comment (AlekSi) [inline nginx.conf:186]: That actually works, neat

File: nginx.conf [hunk 3]
```diff
@@ -36,6 +36,10 @@
 
     client_max_body_size 10m;
 
+    # This root stores static content for general pmm-server purposes.
+    # Ex.: local-rss.xml - contains Percona's news when no internet connection.
+    root /usr/share/pmm-server/static;
```
Comment (BupycHuk) [inline nginx.conf:41]: we can use alias for this to avoid affecting other requests.

Comment (BupycHuk) [reply nginx.conf:41]: here https://github.com/percona/pmm-server/pull/217/files#diff-daaf2fceca49ed034eace91c4e875653R188

Comment (askomorokhov) [reply nginx.conf:41]: no, the alias does not here.

File: nginx.conf (hunk 1 above)
Comment (askomorokhov) [reply nginx.conf]: Changed `root /usr/share/pmm-server` to `root /usr/share/pmm-server/static`

File: nginx.conf [hunk 4]
```diff
@@ -172,11 +172,22 @@
       proxy_set_header Connection "";
     }
 
-    # proxy requests to the Percona news rss feed
+    # This localtion stores static content for general pmm-server purposes.
+    # Ex.: local-rss.xml - contains Percona's news when no internet connection.
+    location /static {
```
Comment (AlekSi) [inline nginx.conf:177]: But please check that this does not overlap with some Grafana route

Comment (askomorokhov) [reply nginx.conf:177]: @AlekSi, it is not.
But good point, now, I'd change to `/pmm-static`?

**Pull request merged**